lib/src/handler.dart

```diff
@@ -7,4 +7,9 @@ import 'dart:async';
 import 'request.dart';
 import 'response.dart';
 
+/// The signature of a function which handles a [Request] and returns a
+/// [Future<Response>].
+///
+/// A [Handler] may receive a request directly for a HTTP resource or it may be
+/// composed as part of a larger application.
```

Comment:
It's not clear what it means to receive a request "directly". Maybe, "A [Handler] may call an underlying HTTP implementation, or it may wrap another [Handler] or a [Client]."